### PR TITLE
feat(oui-datagrid): add datagrid customization

### PR DESCRIPTION
**Pull request by neolitec**
SCDC-744

PR Doc: https://github.com/ovh-ux/ovh-ui-kit-documentation/pull/115
PR Style: https://github.com/ovh-ux/ovh-ui-kit/pull/214

Staging: http://feature_datagrid_customize_staging.ui-kit.ovh//#!/oui-angular/datagrid

### CODE REVIEW

File: packages/oui-datagrid/src/datagrid.controller.js [hunk 1]
```diff
@@ -146,16 +156,33 @@ export default class DatagridController {
...
         }
 
         if (this.extraTopElements.length) {
-            this.hasExtraTopContent = true;
             this.extraTopCompiledTemplate = this.$compile(`<div>${this.extraTopElements[0].innerHTML}</div>`);
+            this.hasExtraTopContent = true;
         }
 
-        this.columns = builtColumns.columns.filter(column => !column.hidden);
+        console.log(this.columnsParameters);
```
Comment (AxelPeter) [inline packages/oui-datagrid/src/datagrid.controller.js]: There is a console here to remove

File: packages/oui-datagrid/src/datagrid.controller.js [hunk 2]
```diff
@@ -146,16 +156,33 @@ export default class DatagridController {
...
+                    name: column.name
+                });
+                console.log(customColumn, column.name);
+                if (customColumn) {
+                    column.hidden = customColumn.hidden;
+                }
+                return column;
+            });
+
+        console.log(this.availableColumns);
```
Comment (AxelPeter) [inline packages/oui-datagrid/src/datagrid.controller.js]: Here too

File: packages/oui-datagrid/src/datagrid.controller.js [hunk 3]
```diff
@@ -146,16 +156,33 @@ export default class DatagridController {
...
 
-        this.columns = builtColumns.columns.filter(column => !column.hidden);
+        console.log(this.columnsParameters);
+
+        this.availableColumns = angular.copy(builtColumns.columns)
+            .map(column => { // Override default with custom columns
+                const customColumn = find(this.columnsParameters, {
+                    name: column.name
+                });
+                console.log(customColumn, column.name);
```
Comment (AxelPeter) [inline packages/oui-datagrid/src/datagrid.controller.js]: And a last one here :)

Comment (neolitec) [reply packages/oui-datagrid/src/datagrid.controller.js]: Oops

Comment (neolitec) [reply packages/oui-datagrid/src/datagrid.controller.js]: BTW, you can easily make build fail with that on CDS if you want.

**Pull request merged**